sidecar/register/healthchecker.go

```diff
@@ -71,34 +72,35 @@ func (checker *HealthChecker) Stop() {
 	checker.stop <- struct{}{}
 }
 
-func (checker *HealthChecker) report() {
+// maintainRegistration
+func (checker *HealthChecker) maintainRegistration() {
 	wasHealthy := false // Initialize as unhealthy so that we register on the first successful health check
 
-	statusChan := make(chan HealthStatus, len(checker.checks))
+	statusChan := make(chan healthcheck.Status, len(checker.checks))
 	for _, healthCheck := range checker.checks {
 		healthCheck.Start(statusChan)
 	}
 
 	// Set of health checks that have most recently reported unhealthy statuses.
-	unhealthyStatuses := make(map[HealthCheck]interface{})
+	unhealthyStatuses := make(map[healthcheck.Check]interface{})
```

Comment:
To avoid the liability, I switched to an implementation of the health checker similar to your suggestion.